src/project-css/app/Http/Controllers/DataViewController.php

```diff
@@ -91,4 +99,120 @@ public function show(Request $request, $curTable, $curId){
                             ->with('tblId',$curTable);
   }
 
+  public function search(Request $request, $curTable, $search = NULL, $page = 1){
+    // test for the validity of curtable
+    if(!$this->isValidTable($curTable)){
+      return redirect()->route('home')->withErrors(['Table id is invalid']);
+    }
+
+    // Get the table entry in meta table "tables"
+    $curTable = Table::find($curTable);
+    if(!$curTable->hasAccess){
+      return redirect()->route('home')->withErrors(['Table is disabled']);
+    }
+
+    if ($search == NULL){
+      $search = $request->input('search');
+    }
+
+    $string_helper = new customStringHelper();
+    $srchStrng = $string_helper->cleanSearchString($search);
+
+    // set records per page
+    $perPage = 30;
+
+    // retrieve the column names
+    $clmnNmes = DB::getSchemaBuilder()->getColumnListing($curTable->tblNme);
+
+    // query sorted by revelancy score
+    $query = DB::table($curTable->tblNme)
+            ->whereRaw("match(srchindex) against (? in boolean mode)", [$srchStrng])
+            ->orderBy('score', 'desc')
+            ->offset($page-1 * $perPage)
+            ->limit($perPage);
+
+    $rcrds = $query
+            ->get(['*', DB::raw("MATCH (srchindex) AGAINST ('".$srchStrng."') AS score")]);
+
+    $rcrdsCount = count($rcrds);
+
+    // if fulltext search yeilds no results try a basic like search
+    // if ($rcrdsCount == 0) {
+    //
+    //   $offsetNum = ($page == 1) ? 0 : ($page-1) * $perPage;
+    //
+    //   // strip characters used by the fulltext boolean search
+    //   //$srchStrng = preg_replace('/[+-]/', '', $srchStrng);
+    //
+    //   $query = DB::table($curTable->tblNme)
+    //           ->where('srchindex', 'like', '%' . $srchStrng . '%')
+    //           ->orderBy('id', 'asc')
+    //           ->offset($offsetNum)
+    //           ->limit($perPage);
+    //
+    //   $rcrds = $query
+    //           ->get();
+    //
+    //   $rcrdsCount = count($rcrds);
+    // }
+
+    // if last query returned exactly 30 items
+    // we assume that their are additional pages
+    // so we set $lastPage to $page + 1
+    $lastPage = ($rcrdsCount == $perPage) ? $page + 1 : $page;
+
+
+    return view('user.search')->with('rcrds', $rcrds)
```

Comment:
I'm told that this produces a bug according to the scruitinzer 

The method `with` does only exist in `Illuminate\View\View`, but not in `Illuminate\Contracts\View\Factory`